### DISCUSSION TITLE
ConstantDeclaration and FieldDeclaration use a different way of indicating modifiers to other productions - why?

### DISCUSSION THREAD

**Discussion by vsajip** [javacc21/javacc21#215] [Q&A] [answered]
Why is the `Modifiers` production not being used in these productions?
https://github.com/javacc21/javacc21/blob/af739f35c3a15fba3d21f4de579b4ef408351987/examples/csharp/CSharp.javacc#L483-L502

**Comment by vsajip** [comment]
Also, this:
https://github.com/javacc21/javacc21/blob/af739f35c3a15fba3d21f4de579b4ef408351987/examples/csharp/CSharp.javacc#L513-L515

**Comment by revusky** [comment, marked as answer]
I don't know offhand why it doesn't use Modifiers. Once we have permissibleModiers working, it should use that, I suppose. It may be that I wrote that part of the grammar at an early stage, before I had the permissibleModifiers scheme even settled on the Java side. I started trying to write the CSharp grammar quite a while ago but then took it up very seriously again at the beginning of this year.
